### PR TITLE
Fix Fsharp template

### PR DESCRIPTION
**Pull request by TheAngryByrd**
Fixed issues I had while testing the F# template

### CODE REVIEW

File: templates/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp.nuspec [hunk 1]
```diff
@@ -9,7 +9,6 @@
     <tags>benchmark;benchmarking;performance;template</tags>
     <iconUrl>https://raw.githubusercontent.com/dotnet/BenchmarkDotNet/master/docs/guide/logo.png</iconUrl>
     <projectUrl>https://benchmarkdotnet.org</projectUrl>
-    <license type="expression">MIT</license>
```
Comment (TheAngryByrd) [inline templates/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp.nuspec]: I couldn't get the package to create without removing this

File: templates/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp/content/BenchmarkConfig.fs [hunk 2]
```diff
@@ -1,20 +1,17 @@
-open BenchmarkDotNet.Analysers;
-open BenchmarkDotNet.Attributes;
-open BenchmarkDotNet.Columns;
-open BenchmarkDotNet.Configs;
-open BenchmarkDotNet.Exporters;
-open BenchmarkDotNet.Exporters.Csv;
-open BenchmarkDotNet.Jobs;
-open BenchmarkDotNet.Loggers;
+module Configs
```
Comment (TheAngryByrd) [inline templates/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp/content/BenchmarkConfig.fs:1]: F# files require either a namespace or a module at the start

File: templates/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp/content/BenchmarkConfig.fs [hunk 3]
```diff
@@ -1,20 +1,17 @@
...
+
+type BenchmarkConfig() as self =
 
-type BenchmarkConfig() =
     // Configure your benchmarks, see for more details: https://benchmarkdotnet.org/articles/configs/configs.html.
-     ManualConfig
-            .Create(DefaultConfig.Instance)
-            .With(Job.ShortRun.With(Runtime.Mono))
-            .With(Job.ShortRun.With(Runtime.Core))
+    inherit ManualConfig()
```
Comment (TheAngryByrd) [inline templates/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp/content/BenchmarkConfig.fs:11]: Fixed the config class

File: templates/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp/content/_BenchmarkName_.fs [hunk 4]
```diff
@@ -1,9 +1,11 @@
...
-open BenchmarkDotNet.Attributes;
+﻿module Benchmark 
+
+open System
+open BenchmarkDotNet
+open BenchmarkDotNet.Attributes
 
 #if config
-[<Config(typeof(BenchmarkConfig))>]
+[<Config(typedefof<Configs.BenchmarkConfig>)>]
```
Comment (TheAngryByrd) [inline templates/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp/content/_BenchmarkName_.fs:8]: Needed to use `typedefof<>` here

File: templates/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp/content/_BenchmarkProjectName_.fsproj [hunk 5]
```diff
@@ -14,4 +14,10 @@
     <PackageReference Include="BenchmarkDotNet" Version="0.1*" />
     <PackageReference Include="BenchmarkDotNet.Diagnostics.Windows" Version="0.1*" />
   </ItemGroup>
+  <ItemGroup Condition="'$(config)' == true">
```
Comment (TheAngryByrd) [inline templates/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp/content/_BenchmarkProjectName_.fsproj:17]: F# requires adding the files in order to a fsproj file.

**Review by TheAngryByrd — COMMENTED** [review]
Added specific comments to why changes were made

File: templates/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp.nuspec (hunk 1 above)
Comment (CodeTherapist) [reply templates/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp.nuspec]: Which nuget version did you use?

**Comment by CodeTherapist** [comment]
Thanks for your work 👍 I appreciate always contributions to my work - since it's a great way to learn!
I would like to understand why my `dotnet build` command was successful. Could you please ellaborate why that could be and which version do you use of Visual Studio / msbuild / dotnet sdk?

File: templates/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp.nuspec (hunk 1 above)
Comment (TheAngryByrd) [reply templates/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp.nuspec]: NuGet Version: 4.8.2.5835

**Comment by TheAngryByrd** [comment]
So `dotnet build` would have been successful but you would have has issues running benchmark since no files were listed in the fsproj. Unfortunately there's [no good official docs](https://github.com/dotnet/docs/issues/11393) on this subject but the gist of it is fsproj files require fs files to be listed as `Compile` in an `ItemGroup`. Also if a file references another `type` or `value` in another file, (such as the benchmark referencing the config class) that file must be above listed above it in the file order.

For reference `dotnet info`

```
dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   2.2.203
 Commit:    e5bab63eca

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.14
 OS Platform: Darwin
 RID:         osx.10.14-x64
 Base Path:   /usr/local/share/dotnet/sdk/2.2.203/

Host (useful for support):
  Version: 2.2.5
  Commit:  0a3c9209c0
```

**Comment by TheAngryByrd** [comment]
Without the files in the fsproj:

```bash
$ dotnet benchmark ./bin/Release/netstandard2.0/benchmark-template-test.dll
No benchmarks to choose from. Make sure you provided public non-sealed non-static types with public [Benchmark] methods.
```

With the files in the fsproj:

```
$ dotnet benchmark ./bin/Release/netstandard2.0/benchmark-template-test.dll

Available Benchmark:
  #0 MyBenchmark


You should select the target benchmark(s). Please, print a number of a benchmark (e.g. '0') or a contained benchmark caption (e.g. 'MyBenchmark'):
If you want to select few, please separate them with space ` ` (e.g. `1 2 3`)
You can also provide the class name in console arguments by using --filter. (e.g. '--filter *MyBenchmark*'):
```

File: templates/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp.nuspec (hunk 1 above)
Comment (CodeTherapist) [reply templates/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp.nuspec]: Not sure what we should do about that - last recommended [version is 5.0.2](https://www.nuget.org/downloads).

Comment (TheAngryByrd) [reply templates/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp.nuspec]: This was version mono 5.18 installed on my machine, 5.20 is out so I'll download that and try it out.

Comment (TheAngryByrd) [reply templates/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp/BenchmarkDotNet.BenchmarkProjectTemplate.FSharp.nuspec]: Oh I had to use `nuget update -self`. Now it works. And I just reverted this.

**Pull request merged**